### PR TITLE
Fixing issue #31 : having modular code

### PR DESCRIPTION
**Pull request by AlexandreCGithub**
# Issue Number:

Fix issue #31 

# Description:

Placing the axios post request in a separated file

# How to test:

launch the server
check that the requesting works as usual

# Other notes:

What about this ? Since it was talked about during the last meeting

### CODE REVIEW

File: lib/axios-post.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import axios from 'axios';
+import { QueryPlan } from "@/lib";
+
+export async function postData<QueryPlan>(url: string, payload: any, username: string, password: string): Promise<QueryPlan> {
```
Comment (tibo-mllr) [inline lib/axios-post.ts:4]: No need to make it generic here

File: app/submit-query/page.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 "use client";
 import { QueryPlan } from "@/lib";
-import axios from "axios";
+import { postData } from "@/lib/axios-post";
```
Comment (tibo-mllr) [inline app/submit-query/page.tsx:3]: Maybe add it in the `index.ts`?

**Review by tibo-mllr — CHANGES_REQUESTED** [review]
You still don't have the linter from the `.vscode/extensions`? The pipeline failed because of formatting

File: lib/axios-post.ts (hunk 1 above)
Comment (AlexandreCGithub) [reply lib/axios-post.ts]: What exactly ?

Comment (tibo-mllr) [reply lib/axios-post.ts]: The type

File: app/submit-query/page.tsx (hunk 2 above)
Comment (tibo-mllr) [reply app/submit-query/page.tsx:3]: (And then merge the two imports)

File: lib/index.ts [hunk 3]
```diff
@@ -1 +1,2 @@
 export * from "./queryPlan";
+export * from "./axios-post";
```
Comment (tibo-mllr) [inline lib/index.ts:2]: In alphabetical order please? :face_holding_back_tears:

File: lib/axios-post.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { QueryPlan } from "@/lib";
+import axios from "axios";
+
+export async function postData(
+  url: string,
+  payload: { mdx: string },
+  username: string,
+  password: string,
+): Promise<QueryPlan> {
+  try {
```
[the remark this answers is not in the document]
Comment (tibo-mllr) [reply lib/axios-post.ts:10]: (Not sure though)

[the remark this answers is not in the document]
Comment (AlexandreCGithub) [reply lib/axios-post.ts:10]: Doesn't work if I remove the import, I don't see how to import it directly

[the remark this answers is not in the document]
Comment (AlexandreCGithub) [reply lib/axios-post.ts:10]: Should I continue to throw the error to page.tsx ?

[the remark this answers is not in the document]
Comment (tibo-mllr) [reply lib/axios-post.ts:10]: Oh yes, right, so you don't need a catch/try at all here!

[the remark this answers is not in the document]
Comment (AlexandreCGithub) [reply lib/axios-post.ts]: I'm using .catch in the function file but what about page.tsx if there is an error ? To choose between setResponse and setError

[the remark this answers is not in the document]
Comment (AlexandreCGithub) [reply lib/axios-post.ts]: Should be ok

**Pull request merged**